core/graphelier-service/api/hndlrs/orderhandler.go

```diff
@@ -0,0 +1,66 @@
+package hndlrs
+
+import (
+	"encoding/json"
+	"errors"
+	"graphelier/core/graphelier-service/models"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/gorilla/mux"
+)
+
+// FetchOrderInfo returns information about an order using its ID and timestamp
+func FetchOrderInfo(env *Env, w http.ResponseWriter, r *http.Request) error {
+	params := mux.Vars(r)
+	instrument := params["instrument"]
+	timestamp := params["timestamp"]
+	orderID := params["orderID"]
+
+	intTimestamp, err := strconv.ParseUint(timestamp, 10, 64)
+	if err != nil {
+		return StatusError{400, err}
+	}
+	intOrderID, err := strconv.ParseInt(orderID, 10, 64)
+	if err != nil {
+		return StatusError{400, err}
+	}
+	SODTimestamp := getClosestSODTimestamp(intTimestamp)
+	EODTimestamp := getClosestEODTimestamp(intTimestamp)
+	messages, err := env.Connector.GetSingleOrderMessages(instrument, SODTimestamp, EODTimestamp, intOrderID)
+	if err != nil {
+		return StatusError{500, err}
+	}
+	if len(messages) == 0 {
+		return StatusError{404, errors.New("Order was not found")}
+	}
+	orderInfoBuilder := &models.OrderInfoBuilder{}
+	orderInfoBuilder.
+		WithID(intOrderID).
+		WithInstrument(instrument).
+		WithMessages(&messages).
+		WithTimestamp(intTimestamp)
+	orderInfo := orderInfoBuilder.Build()
+	w.WriteHeader(http.StatusOK)
```

Comment:
StatusOK is implicitly added when writing data to the response body as part of the `json.Encoder.Encode` call. Furthermore, if the encoding fails the StatusOK should not be written to the request's header, and writing the StatusOK after the encoding has succeeded results in a `superfluous write to headers` warning. I have removed these `WriteHeader` calls from other handlers in #152 for that reason and I think we should do the same here.